exercises/1-User_Mode/challenge_101/test/test_101_challenge.py

```diff
@@ -0,0 +1,131 @@
+        exp_err_list = []  # List of expected stderr output
+
+        # SETUP INPUT
+        # Handled signals
+        for sig_num in range(20, 31):
+            sig_res_list.append(SignalResults(sig_num, False))
+            exp_err_list.append(f'Ignoring signal: {sig_num}')
+        # Non-handled signals
+        sig_res_list.append(SignalResults(9, True))
+
+        # RUN IT
+        self.check_signals(sig_res_list)
+        self.expect_stderr(exp_err_list)
+        self.run_test()
+
+    def test_boundary_03(self):
+        """Smallest valid signal: SIGHUP."""
+        self.check_signals([SignalResults(1, False)])
+        self.expect_stderr(['Ignoring signal: 1'])
+        self.run_test()
+
+    def test_boundary_04(self):
+        """Largest valid signal: SIGRTMAX."""
+        self.check_signals([SignalResults(64, False)])
+        self.expect_stderr(['Ignoring signal: 64'])
+        self.run_test()
+
+
+class Challenge101Special(Challenge101Test):
+    """Defines special test cases."""
+
+    def test_special_01(self):
+        """Start, SIGSTOP, SIGCONT, then SIGKILL.
+
+        SIGSTOP pauses the process.  SIGCONT wakes it back up.  SIGKILL kills it.
+        """
+        self.check_signals([SignalResults(19, False), SignalResults(18, False),
+                            SignalResults(9, True)])
+        self.expect_stderr(['Ignoring signal: 18'])
+        self.run_test()
+
+    def test_special_02(self):
+        """Check the dynamic relocation records.
+
+        Verifies that sigaction() is being called instead of signal().
+
+        From the man page (see: man 2 signal):
+            The  behavior of signal() varies across UNIX versions, and has also varied
+            historically across different versions of Linux.
+            Avoid its use: use sigaction(2) instead.
+        """
+        # LOCAL VARIABLES
+        cmd_list = ['objdump', '-R', self._full_challenge_bin]  # Command to execute
+        command = ' '.join(cmd_list)                            # Human readable command
+        std_out = ''                                            # stdout from command
+        std_err = ''                                            # stderr from command
+
+        # PREPARE
+        self._build_challenge_bin()
+
+        # GET INFORMATION
+        try:
+            (std_out, std_err) = execute_subprocess_cmd(cmd_list)
+        except RuntimeError as err:
+            self.fail(self._test_error.format(str(err)))
+        else:
+            if std_err:
+                self.fail(self._test_error.format(f'Command: {command} produced '
+                                                  f'error {std_err}'))
+            if 'signal' in std_out:
+                self._add_test_failure('signal is essentially deprecated.  See `man 2 signal`')
+            if 'sigaction' not in std_out:
+                self._add_test_failure("You don't seem to be making the right system call yet.")
+
```

Comment:
Add a special test to verify the code isn't using non-async-signal-safe functions.

Reply:
Special 03 added.